File: Data Structures & Algorithms/longest-consecutive-sequence/submission-15.cpp
#include <cmath>
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int c =0;
        unordered_set<int> s(nums.begin(), nums.end());
        for (int r = 0;r<nums.size();r++){
            if (s.count(nums[r]-1) != 1){
                int n = nums[r]+1;
                int count=1;
                while(s.count(n++))
                    count++;
                c = max(c,count);
            }
        }
        return c;
    }
};

File: Data Structures & Algorithms/max-water-container/submission-15.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        l = 0
        r = len(heights) - 1
        m = 0
        while l < r:
            if heights[l] < heights[r]:
                m = max(m, heights[l]*(r-l))
                l+=1
            else:
                m = max(m, heights[r]*(r-l))
                r-=1
        return m

            